MonkeyWrench.Web.UI/EditLane.aspx.cs

```diff
@@ -705,7 +706,16 @@ protected void cmdSave_Click (object sender, EventArgs e)
 		lane.enabled = chkEnabled.Checked;
 		lane.additional_roles = txtRoles.Text;
 		lane.priority = Int32.Parse (lstPriority.SelectedItem.Value);
-		lane.is_protected = chkProtected.Checked;
+		lane.is_protected = chkProtected.Checked;
+
+		// Don't ever pull more than 100 commits
+		try {
+			int commits = Int32.Parse (txtFetchCommits.Text);
+			lane.max_commits_to_fetch = commits >= 100 ? 100 : commits;

```

Comment:
I think that if you want to shoot yourself in the foot by setting `max_commits_to_fetch` to a terrible number, wrench shouldn't stop you :smile: Mostly because the default is sane, you have to explicitly do something (which is quite obviously bad).

Unless you really want this: my scenario (which I've used in the past) is to create a new lane, build something + add a test that show something's broken (and which you know at some point worked), and then just schedule a huge number of commits and let wrench churn through finding the build that broke the test. `git bisect` might usually be a better option, but it might not always be possible to do a `git bisect`.